jdk-annotations/astubx-generator-cli/src/test/java/com/uber/nullaway/jdkannotations/AstubxTest.java

```diff
@@ -0,0 +1,444 @@
+          "import org.jspecify.annotations.Nullable;",
+          "@NullMarked",
+          "public class Generic<T> {",
+          " public String getString(@Nullable T t) {",
+          "   return t.toString();",
+          " }",
+          "}"
+        };
+    ImmutableMap<String, MethodAnnotationsRecord> expectedMethodRecords =
+        ImmutableMap.of(
+            "Generic:java.lang.String getString(T)",
+            MethodAnnotationsRecord.create(
+                ImmutableSet.of(), ImmutableMap.of(0, ImmutableSet.of("Nullable"))));
+    runTest(
+        "Generic.java",
+        lines,
+        expectedMethodRecords,
+        ImmutableMap.of(),
+        ImmutableSet.of("Generic"));
+  }
+
+  @Test
+  public void primitiveTypeReturn() throws IOException {
+    String[] lines =
+        new String[] {
+          "import org.jspecify.annotations.NullMarked;",
+          "import org.jspecify.annotations.Nullable;",
+          "@NullMarked",
+          "public class PrimitiveType {",
+          "    public int multiply(@Nullable Integer num1, @Nullable Integer num2) {",
+          "       if(num1!=null && num2!=null){",
+          "           return num1*num2;",
+          "       }",
+          "       return -1;",
+          "    }",
+          "}"
+        };
+    ImmutableMap<String, MethodAnnotationsRecord> expectedMethodRecords =
+        ImmutableMap.of(
+            "PrimitiveType:int multiply(java.lang.Integer, java.lang.Integer)",
+            MethodAnnotationsRecord.create(
+                ImmutableSet.of(),
+                ImmutableMap.of(0, ImmutableSet.of("Nullable"), 1, ImmutableSet.of("Nullable"))));
+    runTest(
+        "PrimitiveType.java",
+        lines,
+        expectedMethodRecords,
+        ImmutableMap.of(),
+        ImmutableSet.of("PrimitiveType"));
+  }
+
+  @Test
+  public void voidReturn() throws IOException {
+    String[] lines =
+        new String[] {
+          "import org.jspecify.annotations.NullMarked;",
+          "import org.jspecify.annotations.Nullable;",
+          "@NullMarked",
+          "public class VoidReturn {",
+          "    public void printMultiply(@Nullable Integer num1, @Nullable Integer num2) {",
+          "       if(num1!=null && num2!=null){",
+          "           System.out.println(num1*num2);",
+          "       }",
+          "    }",
+          "}"
+        };
+    ImmutableMap<String, MethodAnnotationsRecord> expectedMethodRecords =
+        ImmutableMap.of(
+            "VoidReturn:void printMultiply(java.lang.Integer, java.lang.Integer)",
+            MethodAnnotationsRecord.create(
+                ImmutableSet.of(),
+                ImmutableMap.of(0, ImmutableSet.of("Nullable"), 1, ImmutableSet.of("Nullable"))));
+    runTest(
+        "VoidReturn.java",
+        lines,
+        expectedMethodRecords,
+        ImmutableMap.of(),
+        ImmutableSet.of("VoidReturn"));
+  }
+
+  private void runCompilerWithPlugin(
```

Comment:
I feel like there is a simpler way to run the compiler by creating a `DummyChecker` Error Prone checker and then re-using `CompilationTestHelper`.  See the code in [`NullnessAnnotationSerializerTest`](https://github.com/uber/NullAway/blob/3668f801f0fdb93ef3194d31d27e3d0d8e142439/jdk-javac-plugin/src/test/java/com/uber/nullaway/javacplugin/NullnessAnnotationSerializerTest.java).  Could we use that approach instead of this?